src/services/timeline.rs

```diff
@@ -0,0 +1,78 @@
+use crate::models::{
+    goal::{Day, TimeFilter},
+    slot::Slot,
+    timeline::Timeline,
+};
+
+impl Timeline {
+    pub fn apply_filter(&mut self, _filter: &Option<TimeFilter>) -> Timeline {
+        todo!("apply_filter not implemented");
+    }
+
+    /// Filtering timeline based on before_time and after_time fields in TimeFilter
+    pub fn filter_timing(&mut self, _before_time: Option<usize>, _after_time: Option<usize>) {
```

Comment:
Since we have `apply_filter' exposed towards the outside, the `filter_*` methods can be private?

Reply:
Yes, that is good design, thanks